ffi/dotnet/Devolutions.IronRdp/src/Connection.cs

```diff
@@ -23,17 +23,38 @@ public static class Connection
 
         var serverAddr = ip[0] + ":" + port;
         connector.WithServerAddr(serverAddr);
+        connector.WithDynamicChannelDisplayControl();
         if (factory != null)
         {
             var cliprdr = factory.BuildCliprdr();
             connector.AttachStaticCliprdr(cliprdr);
         }
-
+        
         await ConnectBegin(framed, connector);
         var (serverPublicKey, framedSsl) = await SecurityUpgrade(framed, connector);
         var result = await ConnectFinalize(serverName, connector, serverPublicKey, framedSsl);
         return (result, framedSsl);
     }
+    
+    public static async Task<Written> SingleSequenceStepRead<TStream>(Framed<TStream> frame, ConnectionActivationSequence sequence, WriteBuf buf)
+    where TStream: Stream
+    {
+        buf.Clear();
+
+        var pudHint = sequence.NextPduHint();
+        
+        // Don't remove, NextPduHint is generated and can return null
+        if (null != pudHint)
+        {
+
+            var pdu = await frame.ReadByHint(pudHint);
+
+            return sequence.Step(pdu, buf);
+        }
+
+
+        return sequence.StepNoInput(buf);
+    }
```

Comment:
thought: Is this code duplicated? We may be able to avoid duplication by defining a C# interface.

Reply:
Yeah, but the sequences are generated and does not share interfaces. We certainly could use reflection but that would be harder to maintain and debug in the future.

Reply:
See https://github.com/Devolutions/IronRDP/pull/475